core/include/cubos/core/metrics.hpp

```diff
@@ -0,0 +1,107 @@
+/// @file
+/// @brief Metrics and profiling utilities and macros.
+/// @ingroup core
+
+#pragma once
+
+#include <algorithm>
```

Comment:
```suggestion
```

Just noticed this, do not include this in headers! It eats up compile times because it includes pretty much the whole STL